File: problems/007/007.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    int nth = 10001;
    int n = 1E+6;
    // Using the algorithm of Sieve of Eratosthenes
    int *integers_list = (int*)calloc(n+1, sizeof(int));
    for (int i = 2; (i*i) <= n; i++) {
        if (integers_list[i] == 0) {
            for (int j = (i*i); j <= n; j += i)
                integers_list[j] = 1;
        }
    }
    int primes_n = 0;
    for (int i = 2; i <= n; i++) {
        if (integers_list[i] == 0) {
            primes_n += 1;
            if (primes_n == nth) {
                printf("\n%d\n\n", i);
                break;
            }
        }
    }
    free(integers_list);
    return 0;
}

File: problems/007/007.f90
PROGRAM F007

    ! Copyright 2021 Melwyn Francis Carlo

    IMPLICIT NONE

    INTEGER, PARAMETER :: NTH = 10001, N = 1E+6
    INTEGER :: I, J, PRIMES_N
    LOGICAL, DIMENSION(N+1) :: INTEGERS_LIST

    ! USING THE ALGORITHM OF SIEVE OF EROSTHENES
    INTEGERS_LIST = .TRUE.
    I = 2
    DO WHILE ((I*I) <= N)
        IF (INTEGERS_LIST(I)) THEN
            J = I * I
            DO WHILE (J <= N)
                INTEGERS_LIST(J) = .FALSE.
                J = J + I
            END DO
        END IF
        I = I + 1
    END DO

    PRIMES_N = 0
    DO I = 2, N
        IF (INTEGERS_LIST(I)) THEN
            PRIMES_N = PRIMES_N + 1
            IF (PRIMES_N == NTH) THEN
                GOTO 10
            END IF
        END IF
    END DO

    10 CONTINUE

    PRINT ('(I0)'), I

END PROGRAM F007
